### PR TITLE
Allow configuring docker storage driver in GCE

### PR DESCRIPTION
**Pull request by dashpole**
**What this PR does / why we need it**:
For GCE, allow configuring of the docker storage driver.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by dashpole** [comment]
ok, this should be good to go @yujuhong

**Comment by yujuhong** [comment]
@dashpole verified that the change is good.

/lgtm
/approve no-issue

Need approval from @dchen1107

**Comment by yujuhong** [comment]
Need to cherrypick this back to 1.8, 1.7, and 1.6

**Comment by yujuhong** [comment]
Manually added the "approved" label based on @dchen1107's lgtm.
https://github.com/kubernetes/kubernetes/pull/55512#issuecomment-343634251

**Comment by jpbetz** [comment]
Removing cherrypick label from this PR since PRs for automated cherrypicks exist.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -852,6 +852,12 @@ function assemble-docker-flags {
     docker_opts+=" --live-restore=false"
   fi
 
+  # Override docker storage driver if the environment variable is set
+
+  if [[ -n "${DOCKER_STORAGE_DRIVER:-}" ]]; then
+    docker_opts+=" --storage-driver=${DOCKER_STORAGE_DRIVER}"
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: By the time we reconfigure we have write the docker options we have already loaded the docker images for some of the kube-components:

https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/master.yaml#L31

Changing the storage driver requires reloading of the images. This change breaks GKE (although it's not the only thing currently breaking GKE). I'd like to revert it in https://github.com/kubernetes/kubernetes/pull/55624 so we can start getting signal in GKE again. cc @abgworrall

**Comment by enisoc** [comment]
This has been reverted in #55624.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (dchen1107) [reply cluster/gce/gci/configure-helper.sh]: Ok, we decided to revert this pr.

**Pull request merged**